src/bosh-director/spec/unit/cloud_factory_spec.rb

```diff
@@ -2,56 +2,71 @@
 
 module Bosh::Director
   describe CloudFactory do
-    subject(:cloud_factory) { described_class.new(cloud_planner, parsed_cpi_config) }
+    subject(:cloud_factory) { described_class.new(azs, parsed_cpi_config) }
+
+    let(:azs) { {} }
+
     let(:default_cloud) { Config.cloud }
-    let(:cloud_planner) { instance_double(DeploymentPlan::CloudPlanner) }
     let(:parsed_cpi_config) { CpiConfig::ParsedCpiConfig.new(cpis) }
     let(:cpis) { [] }
     let(:logger) { double(:logger, debug: nil) }
 
     context 'factory methods' do
       let(:cpi_config) { instance_double(Models::Config) }
-      let(:cloud_config) { Models::Config.make(:cloud, content: '--- {"key": "value"}') }
+      let(:cloud_config_content) {
+        { 'azs' => cloud_config_azs}
+      }
+      let(:cloud_config_azs){
+        [{ 'name' => 'z1', 'cpi' => 'cpi_z1'}, { 'name' => 'z2'}]
+      }
+      let(:cloud_config) { Models::Config.make(:cloud, content: YAML.dump(cloud_config_content)) }
       let(:deployment) { instance_double(Models::Deployment) }
       let(:cpi_manifest_parser) { instance_double(CpiConfig::CpiManifestParser) }
-      let(:cloud_manifest_parser) { instance_double(DeploymentPlan::CloudManifestParser) }
-      let(:planner) { instance_double(DeploymentPlan::CloudPlanner) }
 
       before do
         allow(deployment).to receive(:cloud_configs).and_return([cloud_config])
         allow(deployment).to receive(:name).and_return('happy')
-        allow(Api::CloudConfigManager).to receive(:interpolated_manifest).with([cloud_config], 'happy').and_return({})
+        allow(Api::CloudConfigManager).to receive(:interpolated_manifest).with([cloud_config], 'happy').and_return(cloud_config.raw_manifest)
         allow(CpiConfig::CpiManifestParser).to receive(:new).and_return(cpi_manifest_parser)
         allow(cpi_manifest_parser).to receive(:merge_configs).and_return(parsed_cpi_config)
         allow(cpi_manifest_parser).to receive(:parse).and_return(parsed_cpi_config)
         allow(cpi_config).to receive(:raw_manifest).and_return({})
-        allow(DeploymentPlan::CloudManifestParser).to receive(:new).and_return(cloud_manifest_parser)
-        allow(cloud_manifest_parser).to receive(:parse).and_return(planner)
       end
 
       describe '.create_from_deployment' do
         it 'constructs a cloud factory with all its dependencies from a deployment' do
-          expect(described_class).to receive(:new).with(planner, parsed_cpi_config)
-          described_class.create_from_deployment(deployment, [cpi_config])
+          factory = described_class.create_from_deployment(deployment, [cpi_config])
+
+          azs = factory.instance_variable_get(:@azs)
```

Comment:
Aw, testing `instance_variable_get`? We should use public interfaces for testing and not referencing this sort of internal implementation. Can we test via `get_name_for_az` instead?